File: finished/convert-1d-array-into-2d-array.go
package leetcode

// https://leetcode.com/problems/convert-1d-array-into-2d-array

func construct2DArray(original []int, m int, n int) [][]int {
	res := [][]int{}
	if len(original) != m*n {
		return res
	}

	for i := 0; i < m; i++ {
		row := []int{}
		for j := 0; j < n; j++ {
			row = append(row, original[i*n+j])
		}
		res = append(res, row)
	}

	return res
}

File: src/python/finished/convert-1d-array-into-2d-array.py
# https://leetcode.com/problems/convert-1d-array-into-2d-array/

from typing import List


class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        res = []
        if len(original) != m*n:
          return res
       
        for i in range(m):
          row = []
          res.append(row)
          for j in range(n):
            row.append(original[i*n+j])
        return res